docs/reference/components.rst

```diff
@@ -1,10 +1,43 @@
 Components
-**********
+==========
 
 .. include:: /reuse/components-intro.rst
 
-Components employ a `Craft Parts`_ feature called ``partitions``. See
-:ref:`component and partitions<components-and-partitions>` for an explanation
-on how they are related.
+Like snaps, components are packaged a squashfs file. When installed, the
+component is mounted under ``$SNAP/../components/$SNAP_REVISION/`` so
+the host snap can access it.
+
+Types
+-----
+
+A component can have a type for certain purposes. For some component types,
+Snapcraft may place requirements on the contents of a component and snapd
+may have special handling for the component.
+
+Standard
+~~~~~~~~
+
+This is the general use type for components. It should be used when no
+specific type applies. Snapcraft has no requirements or special handling for
+this type.
+
+Kernel-modules
+~~~~~~~~~~~~~~
+
+This type may only be used for :ref:`kernel snaps<kernel-snap-reference>`.
```

Comment:
Something to consider: we use "may" here, but "should" below. Consistency with whether it's obligatory or permitted would improve clarity.

Reply:
Clarified.